src/PhpSpreadsheet/Worksheet/Table.php

```diff
@@ -0,0 +1,407 @@
+<?php
+
+namespace PhpOffice\PhpSpreadsheet\Worksheet;
+
+use PhpOffice\PhpSpreadsheet\Cell\Coordinate;
+use PhpOffice\PhpSpreadsheet\Exception as PhpSpreadsheetException;
+use PhpOffice\PhpSpreadsheet\Worksheet\Table\TableStyle;
+
+class Table
+{
+    /**
+     * Table Name.
+     *
+     * @var string
+     */
+    private $name = '';
+
+    /**
+     * Show Header Row.
+     *
+     * @var bool
+     */
+    private $showHeaderRow = true;
+
+    /**
+     * Show Totals Row.
+     *
+     * @var bool
+     */
+    private $showTotalsRow = false;
+
+    /**
+     * Table Range.
+     *
+     * @var string
+     */
+    private $range = '';
+
+    /**
+     * Table Worksheet.
+     *
+     * @var null|Worksheet
+     */
+    private $workSheet;
+
+    /**
+     * Table Column.
+     *
+     * @var Table\Column[]
+     */
+    private $columns = [];
+
+    /**
+     * Table Style.
+     *
+     * @var TableStyle
+     */
+    private $style;
+
+    /**
+     * Create a new Table.
+     *
+     * @param string $range (e.g. A1:D4)
+     *
+     * @return $this
+     */
+    public function __construct(string $range = '', ?Worksheet $worksheet = null)
+    {
+        $this->setRange($range);
+        $this->setWorksheet($worksheet);
+        $this->style = new TableStyle();
+    }
+
+    /**
+     * Get Table name.
+     *
+     * @return string
+     */
+    public function getName()
+    {
+        return $this->name;
+    }
+
+    /**
+     * Set Table name.
+     *
+     * @return $this
+     */
+    public function setName(string $name)
+    {
+        $this->name = preg_replace('/\s+/', '_', trim($name)) ?? '';
```

Comment:
There's some rules for table names such as max length 255 characters, must begin with a letter, underscore or backslash, can't be the single letter `c` or `r`, can only contain letters, numbers, periods, and underscore, mustn't look like a cell reference, etc.
Should this be validated here?

Reply:
Yes. This should be validated here. I missed it. WIP

Reply:
The other aspect to watch out for here is case-insensitivity: as far as Excel is concerned, a name of `SalesData` is the same as `SALESdata`. I don't have a full list of the rules to hand; but MS must have documented it somewhere.

Reply:
Necessary validations added 3c3d949